### ISSUE TITLE
WebGPU support

### ISSUE THREAD

**Issue by jailln** [iTowns/itowns#2069]
Implement a support of WebGPU (but keep webgl support).

WebGPU [has been released](https://developer.chrome.com/blog/webgpu-release/). It is currently available in Chrome and will be in other browsers soon. It should provide better performances and better developper experience in working with the GPU.

Threejs already started to implement WebGPU. A first step would be to try to use threejs' implementation of webGPU. Then, we will also have to reimplement itowns' shaders and webgl parts with webgpu.

**Comment by jailln** [comment]
From the [following article](https://www.researchgate.net/publication/378843017_WEBGPU_A_NEW_GRAPHIC_API_FOR_3D_WEBGIS_APPLICATIONS):
"Experiments shows that for rendering 3D city models in abrowser environment without using any optimization techniquesWebGPU performs 2.5x-3x better performance than WebGL."

One more reason to try it on :grin:

**Comment by LoicUTC** [comment]
Amazing :)

**Comment by Desplandis** [comment]
Yeah, I agree that we should experiment with `WebGPU` in the short/mid term.

However, it seems that `three.js` will not support `ShaderMaterial` for `WebGPURenderer`, this means we could not write `wgsl` directly... They recommend the use of either the new node material system or use their own `three.js shading language`  (`tsl`) but last time I checked neither were well-documented (or with a few examples)...

**Comment by jailln** [comment]
Ah that's not good news indeed :/ Hopefully they'll document it or provide a better way until we find time to try it
